bgt/datatypes.py

```diff
@@ -81,6 +81,11 @@ def __hash__(self) -> int:
     def __eq__(self, other) -> bool:
         return str(self) == str(other)
 
+    @classmethod
+    def from_str(cls, qid_str: str) -> QuestionID:
+        temp = qid_str.split("-")

```

Comment:
When I have a small number of named arguments, I typically prefer a destructuring assignment in these cases.